app/lib/actions/katello/host/update_system_purpose.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Katello
+    module Host
+      class UpdateSystemPurpose < Actions::EntryAction
+        def plan(host, service_level, purpose_role, purpose_usage, purpose_addons)
+          fail _("Host %s has not been registered with subscription-manager.") % host.name unless host.subscription_facet
+
+          purpose_addon_objects = ::Katello::PurposeAddon.where(name: purpose_addons)
+
+          host.subscription_facet.service_level = service_level
+          host.subscription_facet.purpose_role = purpose_role
+          host.subscription_facet.purpose_usage = purpose_usage
+
+          if host.subscription_facet.purpose_addons.pluck(:id) == purpose_addon_objects.pluck(:id)
```

Comment:
Does refactoring this whole block into the following work?

```
if purpose_addon_objects.any?
  host.subscription_facet.purpose_addons =  purpose_addon_objects
end
```

Reply:
Updated